WalletWasabi/Crypto/ZeroKnowledge/NonInteractive/FiatShamirTransform/Verifier.cs

```diff
@@ -0,0 +1,69 @@
+using System.Linq;
+using System.Collections.Generic;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge.NonInteractive.FiatShamirTransform
+{
+	public delegate VerifyResponse VerifierCommitToNonces(Proof proof);
+
+	public delegate bool VerifyResponse();
+
+	public class Verifier
+	{
+		public Verifier(LinearRelation.Statement statement)
+		{
+			Statement = statement;
+		}
+
+		// Although in principle this could be an interface type, since only linear
+		// relations are needed for our proofs this is specialized to that type.
```

Comment:
Is this comment supposed to be for the constructor? similar to https://github.com/zkSNACKs/WalletWasabi/pull/4309/files#diff-771b458c1ea48799d07234d74ce56dbeR13-R15

Reply:
No.

Reply:
it makes sense either on the constructor or on the property, would have been nice to be consistent but i don't think it's worth changing now, sorry i missed this. i'll do it in #4303 if it makes sense to rewrite comments or move things, there's a good chance of that since we're hoping to avoid keeping the `ToVerifier` and `ToProver` methods